tests/foreman/cli/test_user.py

```diff
@@ -587,3 +587,89 @@ def test_negative_personal_access_token_invalid_date(self, target_sat):
                 action='create',
                 options={'name': token_name, 'user-id': user['id'], 'expires-at': datetime_expire},
             )
+
+
+@pytest.mark.no_containers
+@pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+def test_positive_invalidating_users_tokens(
+    module_target_sat, rhel_contenthost, module_activation_key, module_org, auth_type
+):
+    """Verify invalidating single and multiple users tokens.
+
+    :id: 5db602d4-9c57-4b70-8d46-5323044824e0
+
+    :steps:
+        1. Create an admin user and a non-admin user with "edit_users" and "register_hosts" permission.
+        2. Generate a token with admin user and register a host with it, it should be successful.
+        3. Invalidate the token and try to use the generated token again to register the host, it should fail.
+        4. Invalidate tokens for multiple users with "invalidate-multiple" command, it should invalidate all the tokens for provided users.
+        5. Repeat Steps 2,3 and 4 with non-admin user and it should work the same way.
+
+    :expectedresults: Tokens invalidated cannot be used for registration of hosts.
+
+    :CaseImportance: Critical
+
+    :Verifies: SAT-30385
+    """
+    admin_login = gen_string('alpha')
+    non_admin_login = gen_string('alpha')
+    password = gen_string('alpha')
+    login = admin_login
+    admin_user = module_target_sat.cli_factory.user(
+        {
+            'login': admin_login,
+            'password': password,
+            'admin': '1',
+            'organization-ids': module_org.id,
+        }
+    )
+    user = admin_user
+    # Non-Admin user with "edit_users" permission and "Register hosts" role
+    non_admin_user = module_target_sat.cli_factory.user(
+        {'login': non_admin_login, 'password': password, 'organization-ids': module_org.id}
+    )
+    if auth_type == 'non-admin':
```

Comment:
I think just to simplify this little bit, Can we use just `user` variable here instead of specifying admin_user and non_admin_user explicitely,
```
password = gen_string('alpha')
if auth_type == 'admin':
    user = module_target_sat.cli_factory.user(
        {
            'login': gen_string('alpha'),
            'password': password,
            'admin': '1',
            'organization-ids': module_org.id,
        }
    )
else: 
    user = module_target_sat.cli_factory.user(
        {'login': gen_string('alpha'), 'password': password, 'organization-ids': module_org.id}
    )
    # Create role and assign required permissions here
```

Reply:
Also add teardown step to delete this created user, just to avoid a conflict

Reply:
Creating both admin and non_admin user is required because I want to test the scenario to invalidate both type user tokens with admin user and non-admin user. I would like to keep the original implementation. 
I will add teardown for it.

Reply:
yes, so both type of admin user and non-admin user will be tested with the parametrization separately, while we don't need to have both users at same time, if we keep aside invalidate_multiple scenario, right?

Reply:
We would need both as we have discussed.